packages/dd-trace/src/debugger/devtools_client/snapshot/processor.js

```diff
@@ -0,0 +1,241 @@
+'use strict'
+
+module.exports = {
+  processRawState: processProperties
+}
+
+// Matches classes in source code, no matter how it's written:
+// - Named:                          class MyClass {}
+// - Anonymous:                      class {}
+// - Named, with odd whitespace:     class\n\t MyClass\n{}
+// - Anonymous, with odd whitespace: class\n{}
+const CLASS_REGEX = /^class\s([^{]*)/
+
+function processProperties (props, maxLength) {
+  const result = {}
+
+  for (const prop of props) {
+    // TODO: Hack to avoid periods in keys, as EVP doesn't support that. A better solution can be implemented later
+    result[prop.name.replaceAll('.', '_')] = getPropertyValue(prop, maxLength)
+  }
+
+  return result
+}
+
+function getPropertyValue (prop, maxLength) {
+  // Special case for getters and setters which does not have a value property
+  if ('get' in prop) {
+    const hasGet = prop.get.type !== 'undefined'
+    const hasSet = prop.set.type !== 'undefined'
+    if (hasGet && hasSet) return { type: 'getter/setter' }
```

Comment:
It's purely to add extra meta-data about the given property that might be useful for the user when debugging. You can have a property that only has a setter, one that only has a getter, or one that has both.

Reply:
I see. basically to inform the user why this was not captured.

Maybe this should also include a "noCaptureReason": "property has getter/setter function defined" ?

Reply:
I think seeing that the type is `getter`, `setter` or `getter/setter` without having an option to "fold out" the property is enough for the user to understand that there's nothing more to see here. I don't currently think it adds enough value to warrant the extra data in the JSON. But I can probably be convinced otherwise 😅

Here's an example of how the UI looks without a `noCaptureReason`:

<img width="378" alt="image" src="https://github.com/user-attachments/assets/6d51b30f-70ab-4b01-b989-23670ea7753a">

And here's an example of how the UI looks _with_ a `noCaptureReason`:

<img width="288" alt="image" src="https://github.com/user-attachments/assets/9ae260ef-c7a4-4a31-b6e1-8265408a9ce8">

Here's how I would probably update the code if we wanted to do this:

```js
    if (hasGet && hasSet) return { type: 'getter/setter', notCapturedReason: 'Unable to see value of getter' }
    if (hasGet) return { type: 'getter', notCapturedReason: 'Unable to see value of getter' }
    if (hasSet) return { type: 'setter', notCapturedReason: 'A setter-only property does not have any value' }
```

Reply:
In the spirit of landing this PR, I will leave it as-is for now. But I don't mind updating the code in a follow up PR if you think it's best ☺️